### DISCUSSION TITLE
Is this a good idea to that

### DISCUSSION THREAD

**Discussion by shiminjia** [prisma/prisma#11146] [Q&A] [answered]
For example i have a book table and a comment table
and user can like a book or or a comment.
so i need a like table.

I designed a like table with like.objectId, and made two foreign key to book.id and comment.id,  
but i found it would be foreign key constraint err if there is no same id both in book and comment. 

Book table
```prisma
model Book {
  id                 String          @id @default(cuid())
  likes              Like[]
}
```

Comment table
```prisma
model Comment {
  id        String       @id @default(cuid())
  likes     Like[]
}
```

like table
```prisma
model Like {
  id        String    @id @default(cuid())
  type      LIKE_TYPE @default(BOOK)
  objectId  String?
  like      Boolean   @default(false)
  book      Book?     @relation(fields: [objectId], references: [id], map: "like_book_fk")
  comment   Comment?  @relation(fields: [objectId], references: [id], map:  @"like_comment_fk")
}

enum LIKE_TYPE {
  BOOK
  COMMENT
}
```

Is it a good idea to use one like table to save both Book like record and Comment like record?
What is your opinion to solve this problem?

**Comment by samrathchadha** [comment, marked as answer]
Since all the fields are the same, I don't see much of a problem with this implementation, but it is suggested to have a `ikeId` field for your relations. Docs for that are here: https://www.prisma.io/docs/concepts/components/prisma-schema/relations

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
